x-pack/plugins/security/server/config.ts

```diff
@@ -346,16 +351,19 @@ export function createConfig(
     order: number;
     hasAccessAgreement: boolean;
   }> = [];
+
   for (const [type, providerGroup] of Object.entries(providers)) {
     for (const [name, { enabled, order, accessAgreement }] of Object.entries(providerGroup ?? {})) {
       if (!enabled) {
         delete providerGroup![name];
       } else {
+        const hasAccessAgreement: boolean = !!accessAgreement?.message;
```

Comment:
__optional nit:__ `boolean` is inferred automatically.
```suggestion
        const hasAccessAgreement = !!accessAgreement?.message;
```